### ISSUE TITLE
Images are loaded lazily even though scripting is disabled via uBO (+uM)

### ISSUE THREAD

**Issue by Jee-Hex** [ghacksuserjs/ghacks-user.js#988]
Support for lazy loading was [implemented](https://wiki.developer.mozilla.org/en-US/docs/Mozilla/Firefox/Releases/75) in FF75. 
According to [this](https://bugzilla.mozilla.org/show_bug.cgi?id=1542784#c32) images and iframes are only loaded lazily when JS is enabled because they can otherwise leak a user's approximate scroll position.

**STR:**
1. set uBO to block 1st+3rd-party scripts globally & `no-scripting: * true`
2. set uM to block 1st+3rd-party scripts globally & `noscript-spoof: * true`
3. visit [test site](https://mathiasbynens.be/demo/img-loading-lazy)

**Expected result:**
All images were loaded immediately.

**Actual result:**
Images were loaded lazily.

Images were loaded immediately when JS was disabled via DevTools or setting `javascript.enabled` to `false`. Perhaps users who use extensions to disable JS should consider setting `dom.image-lazy-loading.enabled` to `false`?

**Note:** I have not tested if this affects NS or not.

**Comment by gwarser** [comment]
Looks like documentation does not mention CSP and uBO/uMatrix disable scripting by injecting CSP directive.

**Comment by Thorin-Oakenpants** [comment]
https://whatpr.org/html/3752/images.html#updating-the-image-data

> This is an anti-tracking measure, because if a user agent supported lazy loading when scripting is disabled, it would still be possible for a site to track a user's approximate scroll position throughout a session, by strategically placing images in a page's markup such that a server can track how many images are requested and when.

Unless I'm missing something, I fail to see how a scroll position on a page tracks someone ... "**a site** to track a user's approximate scroll position **throughout a session**" - they can already do that with SSL Session IDs, or cache + ETags or many other ways (zombie persistent local web data, HSTS?). I just don't get it. What relevance does how far down a page you scrolled allow anyone to track you

**Comment by rusty-snake** [comment]
> I fail to see how a scroll position on a page tracks someone

This allows a page without JS to get some information about your usage of a page (read only the header/just scan the article/deeply read a article/…).

TL;DR: Gather metadata on user behaviour when JS is disabled.

**Comment by Thorin-Oakenpants** [comment]
That's nuts. Just because a site (server) knows I scrolled to position 3 out of a possible 5, on a single page ... does not mean it can be used for tracking. It has zero linkability to any other site, let alone page in the same site - there's nothing stable about it

**Comment by rusty-snake** [comment]
The `:hover` mouse position leak is also still present. => If you don't want to be tracked, use sites which don't track you.

https://nakedsecurity.sophos.com/2019/05/09/css-tracking-trick-can-monitor-your-mouse-without-javascript/

**Comment by Jee-Hex** [comment]
> It has zero linkability to any other site, let alone page in the same site - there's nothing stable about it

Unfortunately I am no expert on this but they seem to be referring to 'tracking' in the sense that users' scrolling behavior within a session can be informational for purposes outside of re-identification across sessions? (e.g. % of visitors who read through a long-form article?)

Slightly off-topic: it would seem that Mozilla have [decided](https://bugzilla.mozilla.org/show_bug.cgi?id=1648064#c37) to disable lazy loading in 78.2esr due to issues with printing/print preview and screenshots. The fix (for printing) in FF80 seems to involve forcing all uncached images to load when printing is triggered.

Again this is more of a question if anything, but could this be potentially leaking a user's intent to print a page/ take a screenshot, depending on how far they have scrolled?

**Comment by rusty-snake** [comment]
> potentially leaking a user's intent to print a page

If a site really wants to track this w/o JS, the can add a `@media print { TrackDiv::before { content: url("https://example.com/track/print.png"); } }`.

**Comment by Thorin-Oakenpants** [comment]
FYI: https://gitlab.torproject.org/tpo/applications/tor-browser/-/issues/40056

**Comment by Thorin-Oakenpants** [comment]
> If a site really wants to track this w/o JS, the can add a `@media print { TrackDiv::before { content: url("https://example.com/track/print.png"); } }`.

![mediaquery](https://user-images.githubusercontent.com/16656956/90126644-82850080-dd53-11ea-899f-1040f4c5ee02.png)
